src/ipc/mod.rs

```diff
@@ -20,12 +20,18 @@ pub mod unix;
 
 /// IPC mechanisms must implement this trait.
 pub trait Ipc: 'static + Send {
+    type Addr: Clone + Default + std::cmp::Eq + std::hash::Hash;
     /// Returns the name of this IPC mechanism (e.g. "netlink" for Linux netlink sockets)
     fn name() -> String;
     /// Blocking send
-    fn send(&self, msg: &[u8]) -> Result<()>;
-    /// Blocking listen. Return value is how many bytes were read. Should not allocate.
-    fn recv(&self, msg: &mut [u8]) -> Result<usize>;
+    fn send(&self, msg: &[u8], to: &Self::Addr) -> Result<()>;
+    /// Blocking listen. 
+    ///
+    /// Returns how many bytes were read, and (if using unix sockets) the address of the sender. 
+    /// Other ipc mechanisms are currently implemented as connection pairs and thus leave addr blank.
```

Comment:
nit, I would move this explanation of `Addr` up to the documentation on `pub trait Ipc`.